docs/glossary.rst

```diff
@@ -0,0 +1,152 @@
+==============
+Solar Glossary
+==============
+
+Resources
+==========
+
+Inputs
+--------
+Resource configuration that will be used in actions, handlers and orchestration.
+All known inputs should be provided in meta.yaml ::
+
+    input:
+      apache_ports:
+        value: null
+      fqdn:
+        value: null
+      puppet_modules:
+        value: null
+      role:
+        value: null
+
+Connections
+------------
+Allows to build hierarchy between inputs of several resources,
+parent value will be always used in child while connection is created.
+If connection will be removed - original value of child will be preserved.
+
+Actions
+--------
+Solar wraps deployment code into actions with specific names.
+Several actions of resource are mandatory:
+- run
+- remove
+- update
+
+All actions should be provided in meta.yaml ::
+
+    actions:
+      run: run.pp
+      update: run.pp
+
+Tags
+------
+Used to create arbitrary groups of resources, later this groups will be
+used for different user operations.
+
+Handlers
+=========
+Layer that responsible for action execution and tracking result.
+Currently handler specified in resource meta.yaml and used for all resource
+actions ::
+
+  handler: puppet
+
+Transports
+-----------
+Used in handlers to communicate with managed by solar hosts. List of transports
+should be added to a node. Transports will be added to a resource by means
+of transports id.
+
+Two different types of transports are used: run and sync.
+Run transport - reponsible for running command on remote host.
+Sync transport - uploads required information.
+
+location_id
+------------
+Used in transport layer to find ip address of a node.
+
+transport_id
+------------
+Used to find transports array that will be used for transport selection.
+
+BAT transport
+--------------
+According to preferences solar will choose best available transport for
+file uploading and command execution.
+
+Events
+========
+Used in solar to describe all possible transitions between resources changes.
+Each event allows to specify two points of transitions, condition of this
+transition and type of event.
+
+Right now we are supporting 2 types of events.
+
+1. Dependency
+Inserts edge between 2 changes.
+2. Reaction
+Inserts change specified in reaction and makes edge between parent and child.
+
+Example ::
+
+  type: depends_on
+  parent: nova-db
+  parent_action: run
+  child: nova-api
+  child_action: run
+  state: success // condition
```

Comment:
there is not way to control it with events right now, i will update doc when it will be added